lib/api/funnel.js

```diff
@@ -332,6 +326,47 @@ class Funnel {
     }
   }
 
+  /**
+   * 1/ checking if the requesting user has the right credentials
+   * 2/ send the request itself to the corresponding controller+action
+   * 
+   * @param {Request} request
+   * @param {Function} callback
+   * @returns 
+   */
+  async _preprocessRequest(request, callback) {
+    global.kuzzle.asyncStore.set('REQUEST', request);
```

Comment:
This is set twice, here and line 292 before the triggering of the `request:beforeExecution` event